src/v/cluster/partition_recovery_manager.cc

```diff
@@ -210,7 +210,7 @@ ss::future<log_recovery_result> partition_downloader::maybe_download_log() {
         //
         // The only possible solution here is to discard the exception and
         // continue with normal partition creation process.
-        vlog(_ctxlog.error, "Error during log recovery: {}", err);
+        vlog(_ctxlog.warn, "Error during log recovery: {}", err);
     } catch (...) {
```

Comment:
I'm actually a bit surprised we needed to lower this. I think it means that in the test that failed, we failed to upload a partition manifest while unmounting. @WillemKauf do you recall if we return success to the unmount handler only after writing the manifest?

Reply:
With the subsequent change I'm wondering if the tests pass without this change?

Reply:
[The `flush()` success path only alerts after the partition manifest is uploaded](https://github.com/redpanda-data/redpanda/blob/841ecb06a04f852521dc1eabbd8bd1dd63e95153/src/v/cluster/archival/ntp_archiver_service.cc#L750-L754).

Thanks for pointing this out, I wonder if this log change is suppressing another test bug (and like you mentioned, one that was possibly fixed by the other changes in the PR)

Reply:
No the test still fails. [slack thread on it](https://redpandadata.slack.com/archives/C078LGKDTJ7/p1725366567301979?thread_ts=1725365853.449919&cid=C078LGKDTJ7)

Reply:
It's when adding a new partition to a remounted topic